### ISSUE TITLE
Summarize package-level data in Attribution Notice

### ISSUE THREAD

**Issue by mjherzog** [aboutcode-org/aboutcode-toolkit#328]
When an Inventory data file is based on Package-level information, there are many cases where generating an Attribution entry for each Package may increase the length of the Attribution Notice significantly with redundant information. The most prominent use cases are likely Java projects/Components like Apache Batik or Spring Framework where you many have 10 or more JAR packages with same copyright and license information. We should offer some way to summarize this type of data by project/Component for Attribution Notices. This clearly needs discussion and design.

**Comment by chinyeungli** [comment]
There are 2 ways to accomplish this:
1
User need to create an inventory list which only capture the components which want to be included in the attribution notice. Then use the `attrib` command to generate attribution on *ONLY* the inventory list.
For instance,
about attrib --inventoty [inventory_list] [ABOUT files location] [Output attribution]

2
User can create its own template for this summarization purpose.
Please see the attached files for the sample (Note that the attached samples is "grouped by" license_expression, but same idea for group by component)
[sample.zip](https://github.com/nexB/aboutcode-toolkit/files/2110696/sample.zip)

Following are the commands that I used:
```
about gen --fetch-license 'https://enterprise.dejacode.com/api/v2/licenses/' [API Key] C:\Users\Desktop\test\input.csv C:\Users\Desktop\test

about attrib --template C:\Users\Desktop\test\license_summary.template C:\Users\Desktop\test\328 C:\Users\Desktop\test\output.html
```

The sample.zip contains the input.csv to generate the ABOUT files, the ABOUT files, the custom template for attribution and the attribution output.

@mjherzog / @DennisClark Please check to see if this is enough for the need.

**Comment by DennisClark** [comment]
I will need to test and experiment with templates to produce an attribution notice that summarizes as described in this ticket, and will report the results here.

**Comment by DennisClark** [comment]
@mjherzog   I did some testing with interesting results and observations: 

The template for generating an attribution notice that we normally use is already grouping by component.  If multiple rows in the inventory presented to the `attrib` function have the same component, only one row is printed in the output file using that component value.  Note, however, that the detailed inventory rows sharing that component value (generally referring to multiple files or packages that apply to that same component) are not detailed in the output, only the summary-level component.  That is probably the desired output, but we may need to discuss and confirm that. 

In other words, as the `attrib` software currently works, you simply need to use the same component name (and version) value to summarize multiple rows.  If you don't want to lose that detail in your current component name column, you'll probably need a new column in your software BOM for this purpose.

**Comment by DennisClark** [comment]
Note that if you already have a detailed "filename" or similar column in your Software BOM, you won't lose useful detail by summarizing component+version values for groups of files/packages, but you might need to ensure that the "filename" column is properly populated and understood by the Software BOM readers.

**Comment by chinyeungli** [comment]
@DennisClark I believe we can close this as the template can serve what we need?

**Comment by DennisClark** [comment]
This requirement is met by the template feature of `attrib` .  Closing the issue.
